lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskCommentServiceImpl.java

```diff
@@ -0,0 +1,268 @@
+  }
+
+  TaskComment createTaskComment(TaskComment taskCommentToCreate)
+      throws NotAuthorizedException, TaskNotFoundException, InvalidArgumentException {
+
+    LOGGER.debug("entry to setTaskComment (taskCommentToCreate = {})", taskCommentToCreate);
+
+    TaskCommentImpl taskCommentImplToCreate = (TaskCommentImpl) taskCommentToCreate;
+
+    try {
+
+      taskanaEngine.openConnection();
+
+      taskService.getTask(taskCommentImplToCreate.getTaskId());
+
+      validateNoneExistingTaskCommentId(taskCommentImplToCreate.getId());
+
+      initDefaultTaskCommentValues(taskCommentImplToCreate);
+
+      taskCommentMapper.insert(taskCommentImplToCreate);
+
+    } finally {
+
+      taskanaEngine.returnConnection();
+
+      LOGGER.debug("exit from setTaskComment()");
+    }
+
+    return taskCommentImplToCreate;
+  }
+
+  void deleteTaskComment(String taskCommentId)
+      throws NotAuthorizedException, TaskCommentNotFoundException, TaskNotFoundException,
+          InvalidArgumentException {
+
+    LOGGER.debug("entry to deleteTaskComment (taskComment = {}", taskCommentId);
+
+    String userId = CurrentUserContext.getUserid();
+
+    validateExistingTaskCommentId(taskCommentId);
+
+    try {
+
+      taskanaEngine.openConnection();
+
+      TaskComment taskCommentToDelete = getTaskComment(taskCommentId);
+
+      if (taskCommentToDelete.getCreator().equals(userId)
+          || taskanaEngine.getEngine().isUserInRole(TaskanaRole.ADMIN)) {
+
+        taskCommentMapper.delete(taskCommentId);
+
+        LOGGER.debug("taskComment {} deleted", taskCommentToDelete.getId());
+
+      } else {
+        throw new NotAuthorizedException(
+            NOT_AUTHORIZED + taskCommentToDelete.getCreator() + BUT_CURRENT_USER_IS + userId,
+            userId);
+      }
+
+    } finally {
+      taskanaEngine.returnConnection();
+      LOGGER.debug("exit from deleteTaskComment()");
+    }
+  }
+
+  List<TaskComment> getTaskComments(String taskId)
+      throws NotAuthorizedException, TaskNotFoundException {
+
+    LOGGER.debug("entry to getTaskComments (taskId = {})", taskId);
+
+    try {
+
+      taskanaEngine.openConnection();
+
+      taskService.getTask(taskId);
+
+      List<TaskComment> taskComments =
+          taskCommentMapper.findByTaskId(taskId).stream().collect(Collectors.toList());
+
+      if (taskComments.isEmpty()) {
+        LOGGER.debug("getTaskComments() found no comments for the provided taskId");
+      }
+
+      return taskComments;
+
+    } finally {
+
+      taskanaEngine.returnConnection();
+
+      LOGGER.debug("exit from getTaskComments()");
+    }
+  }
+
+  TaskComment getTaskComment(String taskCommentId)
+      throws TaskCommentNotFoundException, NotAuthorizedException, TaskNotFoundException {
+
+    LOGGER.debug("entry to getTaskComment (taskCommentId = {})", taskCommentId);
+
+    TaskCommentImpl result = null;
+
+    try {
+
+      taskanaEngine.openConnection();
+
+      result = taskCommentMapper.findById(taskCommentId);
```

Comment:
why don't you check for null or empty taskCommentId in this method like in the other methods that have an id as parameter?

Reply:
check added